R/dist_quantiles.R

```diff
@@ -0,0 +1,265 @@
+new_quantiles <- function(q = double(), tau = double()) {
+  arg_is_probabilities(tau)
+
+  vec_cast(q, double())
+  vec_cast(tau, double())
+  stopifnot(length(q) == length(tau))
+  stopifnot(!vec_duplicate_any(tau))
+  if (is.unsorted(tau)) {
+    o <- vec_order(tau)
+    q <- q[o]
+    tau <- tau[o]
+  }
+  if (is.unsorted(q, na.rm = TRUE))
+    rlang::abort("q[order(tau)] produces unsorted quantiles.")
+
+  new_rcrd(list(q = q, tau = tau),
+           class = c("dist_quantiles", "dist_default"))
+}
+
+#' @export
+vec_ptype_abbr.dist_quantiles <- function(x, ...) "dist_qntls"
+#' @export
+vec_ptype_full.dist_quantiles <- function(x, ...) "dist_quantiles"
+
+#' @export
+format.dist_quantiles <- function(x, digits = 2, ...) {
+  q <- field(x, "q")
+  tau <- field(x, "tau")
+  rng <- range(tau, na.rm = TRUE)
+  paste0("[", round(rng[1], digits), ", ", round(rng[2], digits), "]<q-rng>")
+}
+
+
+
+
+#' A distribution parameterized by a set of quantiles
+#'
+#' @param x A vector of values
+#' @param tau A vector of probabilities corresponding to `x`
+#'
+#' @export
+#'
+#' @import vctrs
+#' @examples
+#' dstn <- dist_quantiles(list(1:4, 8:11), list(c(.2,.4,.6,.8)))
+#' quantile(dstn, p = c(.1, .25, .5, .9))
+#' median(dstn)
+#'
+#' # it's a bit annoying to inspect the data
+#' vctrs::vec_data(vctrs::vec_data(dstn[1])[[1]])
+dist_quantiles <- function(x, tau) {
+  if (!is.list(x)) x <- list(x)
+  if (!is.list(tau)) tau <- list(tau)
+
+  x <- as_list_of(x, .ptype = double())
+  tau <- as_list_of(tau, .ptype = double())
+  args <- vec_recycle_common(x = x, tau = tau)
+  qntls <- as_list_of(map2(args$x, args$tau, new_quantiles))
+  new_vctr(qntls, class = "distribution")
+}
+
+
+# format.dist_quantiles <- function(x, ...) {
+#   sprintf("quantiles[%s]", length(x))
+# }
```

Comment:
Was this commented out section left here on purpose? If so, maybe good to document what it's for

Reply:
It was an earlier version that doesn't work correctly. Probably should remove.